# DP: armhf: Fix ffi_call_VFP with no VFP arguments.

--- a/libffi/src/arm/sysv.S
+++ b/libffi/src/arm/sysv.S
@@ -368,6 +368,7 @@ ARM_FUNC_START ffi_call_VFP
 
 	@ Load VFP register args if needed
 	cmp	r0, #0
+	mov	ip, fp
 	beq	LSYM(Lbase_args)
 
 	@ Load only d0 if possible
